### PR TITLE
Committing feature/lqe-203-webchat-component

### PR DESCRIPTION
**Pull request by eamonnprwalsh-zz**

### CODE REVIEW

File: src/components/WebChat/MessageGeneratorUtil.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export const messageHistoryGenerator = () => {
+  var arr = [];
```
Comment (wangx6) [inline src/components/WebChat/MessageGeneratorUtil.js]: const plz

File: src/components/WebChat/MessageGeneratorUtil.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  var len = 100;
```
Comment (wangx6) [inline src/components/WebChat/MessageGeneratorUtil.js]: const plz :)

File: src/components/WebChat/MessageGeneratorUtil.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  for (var i = 0; i < len; i++) {
```
Comment (wangx6) [inline src/components/WebChat/MessageGeneratorUtil.js]:
```
return Array.from({length: 100}).map((e, i)=> {
	// return your record here
});
```

File: package.json [hunk 4]
```diff
@@ -8,7 +8,9 @@
     "@testing-library/user-event": "^12.8.3",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-icons": "^4.2.0",
     "react-scripts": "4.0.3",
+    "react-scroll": "^1.8.2",
```
Comment (wangx6) [inline package.json]: when i open the application, the chatbox doesn't scroll to the bottom

File: src/components/WebChat/WebChat.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  const history = {
+    messages: messageHistoryGenerator(),
+  };
+
+  const [chatHistory, setChatHistory] = useState(history.messages);
+
+  //const divRef = useRef(null);
+
+  /*
+  // Scrolling Parent need to fix
```
Comment (wangx6) [inline src/components/WebChat/WebChat.js]: good thinking man. i dont think we need to use react-scroll to deal with this. 
how about the followings

```
  const chatBox = useRef(null);

  useEffect((event) => {
    chatBox.current.scrollTop = chatBox.current.scrollHeight;
    console.log(chatBox.current.scrollHeight);
  }, []);
```

**Comment by eamonnprwalsh-zz** [comment]
I have addressed the following

// Changed position to fixed
// Fixed scroll
// Added toggle
// Added button for opening W/C
// removed "lqe"
// change message generator
// moved message generator invokation outside main func
// removed capital A&B

@wangx6 Do we use the same pull request now to merge or create a new one?

File: src/components/WebChat/WebChat.css [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  flex-direction: column;
+  position: fixed;
+  right: 0;
+  bottom: 0;
+  bottom: 20px;
+  box-shadow: 5px 5px 20px #fff;
+}
+
+.webchat-panel-inner{
+  flex-direction: column;
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:12]: not sure what this line does. without applying "flex" the "flex-direction" will not take effect anyway.

File: src/components/WebChat/WebChat.css [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  height: 45px;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  background-color:#6e267b;
+  padding: 40px;
+}
+
+.webchat-bar-spacing, .webchat-btn-spacing{
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:25]: i can see that you are trying to manage spacing here, but line 22 also has padding applied. why?

File: src/components/WebChat/WebChat.css [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  padding: 40px;
+}
+
+.webchat-window{
+  flex-direction: column;
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:30]: same as the previous comment regarding flex-direction: column

File: src/components/WebChat/WebChat.css [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  background-color: WhiteSmoke;
+  height: 300px;
+  overflow-y: auto;
+}
+
+.webchat-window, .webchat-bar, .webchat-open__btn, .webchat-message-submit{
+  width: 400px;
+  padding: 20px;
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:38]: padding here and line 46, 47 we have padding again. why?

File: src/components/WebChat/WebChat.css [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+.webchat-dot{
+  height: 20px;
+  width: 20px;
+  background-color: #808080;
+  border-radius: 50%;
+  display: flex;
+  margin-top: 17px;
+}
+
+input{
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:123]: the entire input in your application cud potentially be overwritten by this line.

File: src/components/WebChat/WebChat.css [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+  text-align:left;
+  padding-left: 20px;
+} 
+
+.send-message {
```
Comment (wangx6) [inline src/components/WebChat/WebChat.css:128]: no BEM applied at all. we have to keep consistency in mind all the time :)

File: src/components/WebChat/WebChat.css (hunk 6 above)
Comment (eamonnprwalsh-zz) [reply src/components/WebChat/WebChat.css:12]: The element has another class which has display flex

**Review by wangx6 — APPROVED** [review]

**Pull request merged**